release-26.2: ccl/changefeedccl: TestChangefeedOutdatedCursor failed

ccl/changefeedccl.TestChangefeedOutdatedCursor [failed](https://mesolite.cluster.engflow.com/invocations/default/178dd5f1-2e6e-492f-bb09-2dd71c1f77e7?testReportRun=5&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-26.2 @ [1fd0d177ebd8836ad1354ade07299812216a6ea2](https://github.com/cockroachdb/cockroach/commits/1fd0d177ebd8836ad1354ade07299812216a6ea2):


```
=== RUN   TestChangefeedOutdatedCursor
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestChangefeedOutdatedCursor384321434
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:7104: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedOutdatedCursor384321434
--- FAIL: TestChangefeedOutdatedCursor (0.73s)
=== RUN   TestChangefeedOutdatedCursor/cloudstorage
    helpers_test.go:1578: making server as system tenant
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    helpers_test.go:1654: making cloudstorage feed factory
    changefeed_test.go:7100: expecting CREATE CHANGEFEED FOR TABLE f with cursor = '1774589285717806568.0000000000' to error
    testfeed_test.go:341: creating external connection
    testfeed_test.go:344: ran create external connection
    changefeed_test.go:7100: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:1469
        	            				pkg/ccl/changefeedccl/changefeed_test.go:7100
        	            				pkg/ccl/changefeedccl/helpers_test.go:1876
        	Error:      	"failed to start feed for job 0: pq: internal error while retrieving user account memberships: internal error while retrieving user account: get auth info error: get-login-dependencies: error in retrieving descs between 1774589285.065415239,0, 1774589285.498763965,0: batch timestamp 1774589285.065415239,0 must be after replica GC threshold 1774589285.720440059,0 (r8: /Table/{3-4})" does not contain "could not create changefeed: cursor 1774589285717806568.0000000000 is older than the GC threshold"
        	Test:       	TestChangefeedOutdatedCursor/cloudstorage
--- FAIL: TestChangefeedOutdatedCursor/cloudstorage (0.71s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOutdatedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62117

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -7127,7 +7127,10 @@ func TestChangefeedOutdatedCursor(t *testing.T) {
 		expectErrCreatingFeed(t, f, createChangefeed, expectedErrorSubstring)
 	}
 
-	cdcTestWithSystem(t, testFn, feedTestNoTenants)
+	// Use root to avoid flakes. Non-root connections authenticate against
+	// descriptors that may already be GC'd, so auth fails before the expected
+	// cursor error. See #166880.
+	cdcTestWithSystem(t, testFn, feedTestNoTenants, feedTestUseRootUserConnection)
 }
 
 // TestChangefeedCursorWarning ensures that we show a warning if

